lib/attrsets.nix

```
@@ -333,6 +333,39 @@ rec {
       ) (attrNames set)
     );
 
+   /*
+    Like builtins.foldl' but for Attrsets.
+    Iterates over every nameValuePair in the given AttrSet.
+    The result can be appended to `acc` which gets finally returned.
+    Example
+      reduceAttrs
+      (acc: {name, value}:
+        acc // {
+          sum = acc.sum or 0 + value;
+          names = acc.names or [] ++ [name];
+          };
+      )
+      {}
```

Why not pass the defaults here? You can avoid the `//` and select `or` in the function that way.

```suggestion
      { sum = 0; names = []; }
```

Then your callback becomes:

```
{ sum, names }: name: value: {
  sum = sum + value;
  names = names ++ [name];
}
```